README.md

```diff
@@ -1,19 +1,124 @@
-# Design System
+# DCOS UI Kit
 
-Mesosphere design system prototype
+Mesosphere DCOS resusable UI Kit components
 
-### How to build
+## Getting started
 
-**Requirements**
-1. Install dependencies (Node v8+, NPM 5+)
+Install dependencies (Node v8+, NPM 5+)
 ```bash
 npm i
 ```
 
-**Storybook server**
-1. Run server
+## Development
+
+Start storybook server locally then visit `http://localhost:6006/`
 ```bash
 npm start
 ```
 
-2. Visit `http://localhost:6006/`
+**Linting** <br>
+For linting, use your favorite code editor to enable tslint.
+If you use vscode you can change your workspace settings as following
+```
+{
+  "eslint.enable": false,
+  "jshint.enable": false,
+  "tslint.enable": true,
+  "tslint.jsEnable": false,
+  "tslint.run": "onType"
+}
+```
+
+## New component
+
+To generate a new component run the command
+```
+npm run create:component -- ComponentName
+```
+
+## Unit Testing
+
+```sh
+npm test
+```
+
+Use `test:watch` if you want the tests to run automatically when a file changes:
+
+```sh
+npm run test:watch
+```
+
+You can even pass parameters to the test engine (in this case jest), when you,
+for instance, want to run a single spec, for example, `badge`:
+
+```sh
+npm run test -- --watch badge
+```
+
+**Writing unit tests** <br>
+A recommended reading is [Better Specs](http://www.betterspecs.org/), we put
+[real effort](https://github.com/dcos/dcos-ui/pull/2524) in making sure we
+follow these guidelines. Some of the most common ones to follow:
+
+- Single Expectation test: Every unit test should verify one behavior (and one behavior only).
+- Keep your descriptions concise (bellow 40 chars ideally): One easy way to achieve this one is avoiding using "should" (e.g. "it does not use should"  instead of "it should not be written with should").
+- Create only the data you need: Especially if you have a more complicated scenario, just generate the data that is relevant to that particular case.
+
+For more on this topic, and examples we recommend
+[Better Specs](http://www.betterspecs.org/).
+
+```
+import React from "react";
+import Badge from ".";
+
+describe('Badge', () => {
+  it("render button with children", () => {
+    const badge = mount(<Badge>Wow!</Badge>);
```

Comment:
Is this example functional? Or do we also need to import mount here?

Reply:
I will change this to a real example.

```
it("default", () => {
    expect(renderer
    .create(<Badge>default</Badge>)
    .toJSON()).toMatchSnapshot()
  });
```